### PR TITLE
Imported repos list empty state updates

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

This makes some updates to the empty states of the Imported Repos list.

| No imported repos | No matches | Loaded all results |
|--------|--------|--------|
| image | image | image | 

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at b1260de</samp>

Refactored the repository list page and its components to improve the search and filter functionality, the code structure, and the user interface. Extracted the `RepositoryTable` and `RepoListEmptyState` components to separate files and simplified the `useListConfigurations` hook.

</details>

## Related Issue(s)

Fixes EXP-932

## How to test

* You can [join my org](https://brad-confif73281ba8d.preview.gitpod-dev.com/orgs/join?inviteId=246eef97-0fa0-45d9-b9a3-4aacd08fd2ae) to have > 1 page of results.
* Test loading more. You should see the all results loaded message after you've loaded them all instead the load more button.
* Try searching for `asdf` - you should see the no matching results message
* Check the repo list for your org when there are no imported repos and you should see the empty state.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - brad-confif73281ba8d</li>
	<li><b>🔗 URL</b> - <a href="https://brad-confif73281ba8d.preview.gitpod-dev.com/workspaces" target="_blank">brad-confif73281ba8d.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - brad-configs-empty-state-gha.20403</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-brad-confif73281ba8d%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/repositories/list/RepoListEmptyState.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+type Props = {
+    onImport: () => void;
+};
+export const RepoListEmptyState: FC<Props> = ({ onImport }) => {
+    return (
+        <div className={cn("w-full flex justify-center mt-2 rounded-xl bg-gray-100 dark:bg-gray-800 px-4 py-20")}>
+            <div className="flex flex-col justify-center items-center text-center space-y-4">
+                <Heading2>No repositories yet</Heading2>
+                <Subheading className="max-w-md">
```
Comment (filiptronicek) [inline components/dashboard/src/repositories/list/RepoListEmptyState.tsx:20]: Could we change the color in dark mode for more contrast?
image

File: components/dashboard/src/repositories/list/RepositoryTable.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                    </div>
+                )}
+
+                <div className="mt-4 mb-8 flex flex-row justify-center">
+                    {hasNextPage ? (
+                        <LoadingButton variant="secondary" onClick={onLoadNextPage} loading={isFetchingNextPage}>
+                            Load more
+                        </LoadingButton>
+                    ) : (
+                        hasMoreThanOnePage && <TextMuted>All repositories are loaded</TextMuted>
```
Comment (filiptronicek) [inline components/dashboard/src/repositories/list/RepositoryTable.tsx:104]: We could get really cheesy here one day 😄

**Review by filiptronicek — APPROVED** [review]
:shipit:

File: components/dashboard/src/repositories/list/RepoListEmptyState.tsx (hunk 1 above)
Comment (filiptronicek) [reply components/dashboard/src/repositories/list/RepoListEmptyState.tsx:20]: It applies to this as well 🙏 

image

File: components/dashboard/src/repositories/list/RepoListEmptyState.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { cn } from "@podkit/lib/cn";
+
+type Props = {
+    onImport: () => void;
+};
+export const RepoListEmptyState: FC<Props> = ({ onImport }) => {
+    return (
+        <div className={cn("w-full flex justify-center mt-2 rounded-xl bg-gray-100 dark:bg-gray-800 px-4 py-20")}>
+            <div className="flex flex-col justify-center items-center text-center space-y-4">
+                <Heading2>No repositories yet</Heading2>
```
[the remark this answers is not in the document]
Comment (selfcontained) [reply components/dashboard/src/repositories/list/RepoListEmptyState.tsx]: Good catch. I think I'm going to look at swapping all our headings to the podkit ones to at least get that stuff cleaned up and using the updated components.

File: components/dashboard/src/repositories/list/RepoListEmptyState.tsx (hunk 1 above)
Comment (selfcontained) [reply components/dashboard/src/repositories/list/RepoListEmptyState.tsx:20]: Good catch. I think I'm going to look at swapping all our headings to the podkit ones to at least get that stuff cleaned up and using the updated components.

**Pull request merged**